src/DefaultBuilder/src/WebApplicationOptions.cs

```diff
@@ -64,5 +64,38 @@ internal void ApplyHostConfiguration(IConfigurationBuilder builder)
                 builder.AddInMemoryCollection(config);
             }
         }
+
+        internal void ApplyApplicationName(IWebHostBuilder webHostBuilder)
+        {
+            string? applicationName = null;
+
+            if (ApplicationName is not null)
+            {
+                applicationName = ApplicationName;
+            }
+
+            // We need to "parse" the args here since
+            // we need to set the application name via UseSetting
+            if (Args is not null)
+            {
+                var config = new ConfigurationBuilder()
+                        .AddCommandLine(Args)
+                        .Build();
+
+                applicationName = config[WebHostDefaults.ApplicationKey];
```

Comment:
Should this null check before setting?

Reply:
No, args overrides the property.

Reply:
If args doesn't have ApplicationKey wont it return null?

Reply:
The order is now fixed so this doesn't matter now 😄